docs/Configuration.rst

```diff
@@ -287,6 +287,57 @@ Examples (more below):
 
     $ eb --buildpath=/dev/shm --installpath=/tmp/$USER --disable-debug ...
 
+.. _configuration_show_config:
+
+Overview of current configuration (``--show-config``, ``--show-full-config``)
+-----------------------------------------------------------------------------
+
+To get an overview of the current EasyBuild configuration across all configuration types,
+you can use ``eb --show-config``.
+
+The output will specify:
+
+* any configuration setting for which the current value is different from the default value 
+* a couple of selected important configuration settings (even if they are still set to the default value), i.e.:
+
+  * build path (see :ref:`buildpath`)
+  * install path (see :ref:`installpath`)
+  * path to easyconfigs repository (see :ref:`easyconfigs_repo`)
+  * the robot search path (see :ref:`robot_search_path`)
+  * source path (see :ref:`sourcepath`)
+
+* through which configuration type each setting was defined
+
+  * i.e., default value, configuration file, environment variable or command line argument
+
+Example output::
+
+    $ cat $HOME/.config/easybuild/config.cfg
+    [config]
+    buildpath = /tmp/eb-build
+    
+    $ export EASYBUILD_MODULES_TOOL=Lmod
+    $ export EASYBUILD_OPTARCH=''
+
+    $ eb --show-config --installpath=$HOME/apps --job-cores=4
+    #
+    # Current EasyBuild configuration
+    # (C: command line argument, D: default value, E: environment variable, F: configuration file)
+    #
+    buildpath      (F) = /tmp/eb-build
+    installpath    (C) = /Users/example/apps
+    job-cores      (C) = 4
+    modules-tool   (E) = Lmod
+    optarch        (E) = ''
+    repositorypath (D) = /Users/example/.local/easybuild/ebfiles_repo
+    robot-paths    (D) = /Users/example/easybuild-easyconfigs/easybuild/easyconfigs
+    sourcepath     (D) = /Users/example/.local/easybuild/sources
```

Comment:
all good, useful feature.

an idea that comes to mind: it would be nice, if the output corresponded directly with shell variables, fi.:
`EASYBUILD_MODULES_TOOL=Lmod` etc; OTOH, this just one of the many ways to tune things...